certora/specs/compoundMath.spec

```diff
@@ -1,34 +1,35 @@
 methods {
-    mul(uint256, uint256)   returns (uint256)   envfree
-    div(uint256, uint256)   returns (uint256)   envfree
+    function mul(uint256, uint256)   external returns (uint256)   envfree;
+    function div(uint256, uint256)   external returns (uint256)   envfree;
 }
 
 definition WAD()  returns uint = 10^18;
+definition UINT_LIMIT() returns mathint = 2 ^ 255 * 2;
 
 /// mul ///
 
 rule mulSafety(uint256 x, uint256 y) {
     uint res = mul(x, y);
 
-    assert res == x * y / WAD(); 
+    assert to_mathint(res) == x * y / WAD();
```

Comment:
I understand that know math integers and "solidity integers" are not comparable as is, you need to specify the type conversion is that correct ? And what happens if you try the opposite conversion (math => solidity) ?

Reply:
>And what happens if you try the opposite conversion (math => solidity) ?

I don't think this functionality exists, it would be weird because mathints are obtained from solidity integers

Reply:
`to_mathint` is an internal function in certora?